pydoctor/templatewriter/pages/__init__.py

```
@@ -189,78 +195,81 @@ def namespace(self, obj: model.Documentable) -> Sequence[Union[Tag, str]]:
         return self.docgetter.get(self.ob)
 
-    def children(self):
+    def children(self) -> Sequence[model.Documentable]:
         return sorted(
             (o for o in self.ob.contents.values() if o.isVisible),
             key=objects_order)
 
-    def packageInitTable(self):
+    def packageInitTable(self) -> "Flattenable":
         return ()
 
     @renderer
-    def baseTables(self, request, tag):
+    def baseTables(self, request: object, tag: Tag) -> "Flattenable":
         return ()
 
-    def mainTable(self):
+    def mainTable(self) -> "Flattenable":
         children = self.children()
         if children:
             return ChildTable(self.docgetter, self.ob, children,
                     ChildTable.lookup_loader(self.template_lookup))
         else:
             return ()
 
-    def methods(self):
+    def methods(self) -> Sequence[model.Documentable]:
         return sorted((o for o in self.ob.contents.values()
                        if o.documentation_location is model.DocLocation.PARENT_PAGE and o.isVisible), 
                       key=objects_order)
 
-    def childlist(self):
+    def childlist(self) -> List[Union["AttributeChild", "FunctionChild"]]:
         from pydoctor.templatewriter.pages.attributechild import AttributeChild
         from pydoctor.templatewriter.pages.functionchild import FunctionChild
 
-        r = []
+        r: List[Union["AttributeChild", "FunctionChild"]] = []
 
         func_loader = FunctionChild.lookup_loader(self.template_lookup)
         attr_loader = AttributeChild.lookup_loader(self.template_lookup)
 
         for c in self.methods():
             if isinstance(c, model.Function):
                 r.append(FunctionChild(self.docgetter, c, self.functionExtras(c), func_loader))
-            else:
+            elif isinstance(c, model.Attribute):
                 r.append(AttributeChild(self.docgetter, c, self.functionExtras(c), attr_loader))
+            else:
+                assert False, type(c)
```

Maybe we could raise a `RuntimeError` instead ? saying something like `"unexpected child item in {self.ob.fullName()}"`

---

This is an internal consistency check, so I think an assert is fine here. It checks the types returned by `method()`, which is defined inside the same class. Poor naming though, since it doesn't return methods.